clients/mobile/src/setup/notification/NotificationProvider.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect, ReactNode } from 'react'
+import * as Notifications from 'expo-notifications'
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { SCREENS } from '../constant/screens'
+import { parseJsonData } from '../../utility/jsonParser'
+import { NotificationContext } from './NotificationContext'
+import { useNavigationReady } from './useNavigationReady'
+import { NotificationData } from './NotificationData'
+
+type RootStackParamList = {
+  Home: undefined;
+  ResponseDonationRequest: { notificationData: NotificationData };
+}
+
+export const NotificationProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const [notificationData, setNotificationData] = useState<NotificationData | null>(null)
+  const navigation = useNavigation<NavigationProp<RootStackParamList>>()
+  const waitForNavigationReady = useNavigationReady(navigation)
+
+  useEffect(() => {
+    let isMounted = true
```

Comment:
i struggle to understand

1. why is isMounted a let given you don't seem to be re-assigning it; you're returning a value and setting it to false, which is technically not a reassignment.
2. What isMounted actually does.